### PR TITLE
Violation Table integration with API.

### PR DESCRIPTION
**Pull request by alizenguyen**
Added border changes and bounds adjustment with violation table click.

### CODE REVIEW

File: src/timeline/components/timeline.vue [hunk 1]
```diff
@@ -73,7 +73,9 @@
 		</div>
 	</div>
 	<violations-table
+        :violations="violations"
 		@loadViolations="addErrorsOnLoad" 
+        @resetBounds="resetTimeBoundsFromViolationClick"
```
Comment (deeptailor) [inline src/timeline/components/timeline.vue:78]: Some weird tab differences between these props

File: src/timeline/components/timeline.vue [hunk 2]
```diff
@@ -327,6 +331,12 @@ export default {
             this.violationClicked = value.violationClicked;
 		},
 		onViolationClicked(value) {
+            let violationTime = this.timeFormatter.parse(value.violation.violationTime);
+            let end = violationTime + 20000000;
+            let start = violationTime - 20000000;
```
Comment (deeptailor) [inline src/timeline/components/timeline.vue:336]: Can we define these in a const at the top of the file

File: src/timeline/components/timeline.vue [hunk 3]
```diff
@@ -335,6 +345,16 @@ export default {
 				violators: value.violation.violators,
 			});
 		},
+        resetTimeBoundsFromViolationClick(value) {
+            // let end = this.savedBounds.end;
+            // let start = this.savedBounds.start;
+
+            // if(end && start){
+            //     this.openmct.time.bounds({start, end});
+            // }
```
Comment (deeptailor) [inline src/timeline/components/timeline.vue:354]: We can remove commented code if we dont need it.

File: src/timeline/components/timeline.vue [hunk 4]
```diff
@@ -327,6 +331,12 @@ export default {
             this.violationClicked = value.violationClicked;
 		},
 		onViolationClicked(value) {
+            let violationTime = this.timeFormatter.parse(value.violation.violationTime);
+            let end = violationTime + 20000000;
+            let start = violationTime - 20000000;
+
+            this.openmct.time.bounds({start, end});
+
```
Comment (deeptailor) [inline src/timeline/components/timeline.vue:339]: Some weird tab spacing differences, not sure what happened here.

**Review by deeptailor — CHANGES_REQUESTED** [review]
Looks good, just a few comments.

File: src/timeline/components/timeline.vue (hunk 2 above)
Comment (alizenguyen) [reply src/timeline/components/timeline.vue]: Changed it to use the timeline padding you had already set instead of using 20000000 value.

**Review by deeptailor — COMMENTED** [review]
LGTM!

**Pull request merged**